homeassistant/components/history.py

```diff
@@ -35,6 +35,49 @@ def last_5_states(entity_id):
     return recorder.query_states(query, (entity_id, ))
 
 
+def get_significant_states(
+        start_time, end_time=None, entity_id=None):
+    """
+    Return states changes during UTC period start_time - end_time.
+    """
+    all_states_significant = ('thermostat',)
+    domain_where = "domain in ({})".format(
+        ",".join(["'%s'" % x for x in all_states_significant]))
+    where = "({} or last_changed=last_updated) AND last_updated > ? ".format(
+        domain_where)
+
+    data = [start_time]
+
+    if end_time is not None:
+        where += "AND last_updated < ? "
+        data.append(end_time)
+
+    if entity_id is not None:
+        where += "AND entity_id = ? "
+        data.append(entity_id.lower())
+
+    query = ("SELECT * FROM states WHERE {} "
+             "ORDER BY entity_id, last_updated ASC").format(where)
+
+    states = recorder.query_states(query, data)
```

Comment:
This code till the end of the method seems to be the exact same as the second part of `state_changes_during_period`. Could you extract it into a helper method?